File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/repositories/tenant/UserRepository.java
package com.bittokazi.oauth2.auth.server.app.repositories.tenant;

import com.bittokazi.oauth2.auth.server.app.models.tenant.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, String> {

    Optional<User> findOneByEmail(String email);

    Optional<User> findOneByUsernameIgnoreCase(String name);

    Optional<User> findOneByUsername(String name);

    Optional<User> findOneByImageName(String name);

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/repositories/tenant/UserRepository.kt
package com.bittokazi.oauth2.auth.server.app.repositories.tenant

import com.bittokazi.oauth2.auth.server.app.models.tenant.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, String> {
    fun findOneByEmail(email: String?): Optional<User?>

    fun findOneByUsernameIgnoreCase(name: String?): Optional<User?>

    fun findOneByUsername(name: String?): Optional<User?>

    fun findOneByImageName(name: String?): Optional<User?>
}
